beaker_kernel/lib/types.py

```diff
@@ -2,10 +2,21 @@
 from dataclasses import dataclass, field, MISSING, asdict
 from datetime import datetime, date
 
-typing.TypedDict
-
 DatasourceTypes: typing.TypeAlias = typing.Literal["api", "database", "dataset"]
 
+@dataclass
+class DatasourceAttachment:
+    # user facing name
+    name: str
+
+    # optional - None could be an unsaved new file held in memory but not on disk
+    filepath: typing.Optional[str] = field(default=None)
+    # if None - either zero-byte file or content to load at some point in the future
+    content: typing.Optional[bytes] = field(default=None)
+    # disambiguation for the above field
+    is_empty_file: bool = field(default=False)
+    # TODO: encoding?
+
 @dataclass
 class Datasource:
     uid_or_slug: str
```

Comment:
Please go ahead and rename this variable. It looks like you were using slug in the Biome code, so feel free to make this match if that's the term you prefer. Don't forget to refactor the typescript code to be consistent.